### ISSUE TITLE
Enhancement: menu format option for terminal emulator.

### ISSUE THREAD

**Issue by Thyname** [bbidulock/xde-menu#6]
Currently xde-menu translates 'Terminal=true' in the .desktop file to an xterm command. It would be a nice improvement if there was a menu format option to define the terminal emulator to use.

Thank you.

**Comment by bbidulock** [comment]
Yes, I have been meaning to, but just haven't gotten around to it yet.  I'm a roxterm fan myself.

**Comment by bbidulock** [comment]
I think that you will find that the current release of [xde-menu](/bbidulock/xde-menu/releases/tag/0.10) and [xdg-launch](/bbidulock/xdg-launch/releases/tag/1.9) provide the ability to select the terminal emulation program.  It will also search for an available one if none is specified.

**Comment by Thyname** [comment]
Hello Brian, thank you.

I can set the default terminal emulator for xde-menu with $TERMINAL and it works fine (tested with xfce4-terminal).
But I'm not being able to figure how to set it through .Xresources.

In xde-menu manual there is:
`xde-menu[.WMNAME].termname: TERMINAL-NAME
           Word specifying the name of the terminal emulation program executable to be used to launch, or pass to
           xdg-launch(1), for applications that need terminals.

       xde-menu[.WMNAME].terminal: EXEC-COMMAND
           String specifying the terminal emulation program command to be used to launch, or pass to xdg-launch(1), for
           applications that need terminals.  This command can contain "Exec" line substitutions.

       xde-menu[.WMNAME].termresn: EXEC-COMMAND
           String specifying the terminal emulation program command to be used when a resource name is available and can
           be set to launch, or pass to xdg-launch(1), for applications that need terminals.  This command can contain
           "Exec" line substitutions; however, their "%" must be escaped with an additional "%", and somewhere in the
           command line there must appear a printf(3) style unescaped %s expression to be substituted with the resource
           name.`

I have in ~/.Xresources

`xde-menu.termname: xfce4-terminal
xde-menu.terminal: " -x "`

Would you mind telling me what's wrong?

Edit: maybe WMNAME is missing. I'll try that.

**Comment by Thyname** [comment]
I'm closing this because the original issue is apparently solved. Now xde-menu look for many alternative terminal emulators and uses the first available.
Having, for instance, only xfce4-terminal installed, xde-menu uses it and it works as it should.

**Comment by bbidulock** [comment]
I think you will find that just:

    xde-menu.termname: xfce4-terminal

will do.  [xde-menu](/bbidulock/xde-menu) will fill out the other two as it "knows" about `xfce4-terminal`.  For the other two, when `xfce4-terminal` is set, the default will be like:

    xde-menu.terminal: xfce4-terminal -T "%c" -x 

(Note the trailing space and the lack of double quotes.)  Note that `xfce4-terminal` has no `xde-menu.termresn' resource as it cannot set resource name on the launched terminal window, but, because it properly addresses startup notification completion (copies the `DESKTOP_STARTUP_ID` environment variable into the `_NET_STARTUP_ID` window property on the window or `WM_CLIENT_LEADER` window), this is not an issue.

So, long story short, it was just your syntax on the `xde-menu.terminal` resource that was the problem.

Note also that if you install [xdg-launch](/bbidulock/xdg-launch), [xde-menu](/bbidulock/xde-menu) will use it instead of launching applications itself.  It will pass the terminal settings to [xdg-launch](/bbidulock/xdg-launch) with environment variables.  The purpose of this is that [xdg-launch](/bbidulock/xdg-launch) does a way better job at launching applications with startup notification and other assistance for the window manager than [xde-menu](/bbidulock/xde-menu): which only uses GTK tools to launch.
